UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -94,7 +87,18 @@ private void Shutdown()
         /// </summary>
         private void BrainDecideAction()
         {
-            m_BrainBatcher?.SendBrainInfo(name, m_AgentInfos);
+            m_Communicator?.PutObservations(name, m_AgentInfos);
+            var actions = m_Communicator?.GetActions(name);
```

Comment:
I pulled this logic out of the Communicator to allow brains to fetch actions.

Reply:
This actually changes the behavior to update the actions every time a decision is requested instead of only when every brain has put its observations and a message was sent and received from the communicator

Reply:
How do you allow for the multiple brain scenario? The communicator must still send the messages from all the brains and receive message from all the brains between the PutObservation and the GetAction...

Reply:
It hasn't changed for the multiple brain scenario.  The code was pretty much copy pasted from the batcher into the RpcCommunicator.

Reply:
Not sure what the 👀 emoji means.